src/main/scala/net/snowflake/spark/snowflake/io/StageWriter.scala

```diff
@@ -225,12 +225,23 @@ private[io] object StageWriter {
           format,
           fileUploadResults
         )
-      } else {
+      } else if (params.skipWriteWhenWritingEmptyDataFrame) {
```

Comment:
I don't understand why this design is necessary. My idea is, if a DataFrame has empty partition, then we can directly create a table without invoke COPY.

Reply:
Yes. We can do that.
If the write mode is append, we will `CREATE TABLE if not exist ...`
If the write mode is overwrite, we will execute `CREATE OR REPLAE  table ...`

I will do that.

Reply:
We log only if this parameter is `true`? Don't we still need to create the table?

Reply:
The intention for the `if (params.skipWriteWhenWritingEmptyDataFrame)` is to go to the OLD behaviour if necessary. So we don't create table for it.
By default, it is false, so it will go to `createTable()` in `else` clause

Reply:
let's keep this parameter to play it safe. In the same time, Jira https://snowflakecomputing.atlassian.net/browse/SNOW-330109 is filed to clean-up the internal option in the future.